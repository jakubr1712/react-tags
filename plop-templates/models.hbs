export interface I{{properCase name}}ComponentProps {}