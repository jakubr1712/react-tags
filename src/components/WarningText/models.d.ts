export interface IWarningTextProps {
  message: string;
  color: string;
}
