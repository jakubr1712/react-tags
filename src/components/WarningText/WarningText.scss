.warning-text {
  display: flex;
  align-items: center;
  font-size: 14px;
  color: #a09ea9;

  &__icon {
    margin-right: 8px;
    font-size: 18px;
  }
}
