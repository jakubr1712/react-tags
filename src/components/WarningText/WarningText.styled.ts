import styled from "styled-components";

export const WarningTextContainer = styled.div`
  display: flex;
  align-items: center;
  font-size: 14px;
`;

export const WarningIcon = styled.span`
  margin-right: 8px;
  font-size: 18px;
`;
