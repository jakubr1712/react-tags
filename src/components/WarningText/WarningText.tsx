import React from "react";
import './WarningText.scss';
import {IWarningTextProps} from './models.d';


const WarningText: React.FC<IWarningTextProps> = ({ message, color }) => {
  return (
    <div className="warning-text" style={{ color }}>
      <span className="warning-text__icon">ℹ️</span>
      {message}
    </div>
  );
};

export default WarningText;
