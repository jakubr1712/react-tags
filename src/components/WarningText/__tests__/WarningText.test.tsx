import React from "react";
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import WarningText from "../WarningText";
import { IWarningTextProps } from "../models.d";

describe("WarningText component", () => {
  it("renders properly and matches snapshot", () => {
    const message = "This is a warning message";
    const { container } = render(<WarningText message={message} />);

    expect(container).toMatchSnapshot();
  });

  it("displays the correct message", () => {
    const message = "This is a warning message";
    render(<WarningText message={message} />);

    expect(screen.getByText(message)).toBeInTheDocument();
  });

  it("renders the warning icon", () => {
    const message = "This is a warning message";
    render(<WarningText message={message} />);

    expect(screen.getByText("ℹ️")).toBeInTheDocument();
  });
});
