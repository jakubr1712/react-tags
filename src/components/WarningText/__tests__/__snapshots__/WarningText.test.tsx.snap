// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WarningText component renders properly and matches snapshot 1`] = `
<div>
  <div
    class="warning-text"
  >
    <span
      class="warning-text__icon"
    >
      ℹ️
    </span>
    This is a warning message
  </div>
</div>
`;
